### ISSUE TITLE
Null not possible

### ISSUE THREAD

**Issue by schonhoff** [axlon/laravel-postal-code-validation#23]
Hello,

I get the following error after trying to pass a 'null' to the postal code validation.

```
TypeError
Argument 2 passed to Axlon\PostalCodeValidation\PostalCodeValidator::passes() must be of the type string,
null given, called in /html/vendor/axlon/laravel-postal-code-validation/src/Extensions/PostalCode.php on line 85
```

I fixed it by adding a 
```
'bail', 'required'
```
to my rule set but I think the package should check itself. Maybe you can add an additional test if the given postal code is null and return false. Here my possible workaround:

```
 /**
     * Determine if the given postal code(s) are valid for the given country.
     *
     * @param string $countryCode
     * @param string|null ...$postalCodes
     * @return bool
     */
    public function passes(string $countryCode, ?string ...$postalCodes): bool
    {
       // Added
        if(!isset($postalCodes)) {
           return false;
        }

        if (!$this->supports($countryCode)) {
            return false;
        }

        if (($pattern = $this->patternFor($countryCode)) === null) {
            return true;
        }

        foreach ($postalCodes as $postalCode) {
            if (preg_match($pattern, $postalCode) !== 1) {
                return false;
            }
        }

        return true;
    }
```

Thanks for the work and if I need to check in a pr, just ask :-)

**Comment by axlon** [comment]
Thank you for the bug report @schonhoff. The correct behaviour for receiving `null` should be validation failing, `bail` should not be necessary (except if you want to suppress multiple validation errors on the same input) so this sounds like a bug.

Could you tell me you Laravel and Postal code validation versions?

**Comment by schonhoff** [comment]
Hello,

no problem. I hope I can help you with this information:

`
"name": "axlon/laravel-postal-code-validation",
"version": "v3.1.0",
`

`
"name": "laravel/framework",
"version": "v8.6.0",
`

My current validation rule for my postal code field:
```
['bail', 'required', 'postal_code:DE']
```
As you noticed correctly bail shouldn't be necessary, but currently it is for me.
If I return nothing (null) to the validation

```
$validator = Validator::make($data, $rules, $messages);
if($validator->fails())
```

throws the error message. With the fixed code on my first post I don't get an error and the validation works like you described.

**Comment by schonhoff** [comment]
Just a hint:

on your src\Extensions\PostalCode Class you are using
`
public function validate(string $attribute, ?string $value, array $parameters)
`
on the validate function but in the function passes you are calling there needs to be a string.

`
public function passes(string $countryCode, string ...$postalCodes): bool
`

I guess that this is the error. Maybe it is because of php 7.4 that I'm currently using.

**Comment by axlon** [comment]
@schonhoff Could you review the changes I made in PR linked above? I think they should fix your issue. Note that validation will still pass if the given country has no pattern and null is passed

**Comment by schonhoff** [comment]
@axlon copied the changes to my local system and it seems like it fixes the issue. Thanks for the fast fix!

**Comment by axlon** [comment]
@schonhoff I've merged the changes and tagged it as v3.1.1
